aptos-move/framework/aptos-framework/sources/nonce_validation.move

```diff
@@ -0,0 +1,148 @@
+module aptos_framework::nonce_validation {
+    use aptos_std::table::{Self, Table};
+    use aptos_std::timestamp;
+    use aptos_std::big_ordered_map::{Self, BigOrderedMap};
+    use aptos_std::aptos_hash::sip_hash_from_value;
+    use aptos_std::option;
+    use aptos_std::vector;
+    friend aptos_framework::genesis;
+    friend aptos_framework::transaction_validation;
+
+    const NUM_BUCKETS: u64 = 50000;
+    const MAP_SWITCH_TIME: u64 = 75;
+    const MAX_EXPIRATION_TIME: u64 = 60;
+
+    struct Bucket has store {
+        // Maps NonceKey = (address, nonce) to txn expiration time.
+        // Using multiple maps to facilitate easier garbage collection.
+        nonces: vector<BigOrderedMap<NonceKey, u64>>,
```

Comment:
this seems to be exact 2? also it's not clear how this helps garbage collection

Reply:
Yes. It's exactly 2 big ordered maps.
At any time, only of the maps is considered "active".
We search for the given input (address, nonce) pair in both the big ordered maps.
If the (address, nonce) pair doesn't exist, we insert the entry into only the active map.
Here, I'm switching which of the big ordered map is considered "active" once every 75 seconds.
As we only add entries into the active map, over time, all the entries in the other map would have been inserted over 60 seconds ago. When the last_stored_time of a map is over 60 seconds, we clear the map. This 2 map approach gives us an easy trigger condition on when to do garbage collection. Does it make sense?